libs/api/domains/translations/src/lib/translations.service.ts

```diff
@@ -35,21 +41,24 @@ export class TranslationsService {
 
     const result = await this.contentfulRepository
       .getLocalizedEntries<any>('*', {
-        ['content_type']: 'namespace',
-        select: 'fields.strings,fields.fallback',
+        ['content_type']: 'namespaceJeremyDev', // TODO: replace after review
+        select: 'fields.strings',
         'fields.namespace[in]': namespaces.join(','),
       })
       .catch(errorHandler('getNamespace'))
 
-    const withFallbacks = result?.items?.map(({ fields }) =>
-      mergeWith(
-        {},
-        locale.fallbackCode ? fields.strings[locale.fallbackCode] : {},
-        fields.strings[locale.code],
-        (o, s) => (isEmpty(s) ? o : s),
-      ),
+    return (
+      result?.items?.map(({ fields }) =>
+        fields.strings['is-IS'].reduce(
+          (acc: Record<string, string>, cur: DictArray) => ({
+            ...acc,
```

Comment:
I would suggest replacing the reduce with a simpler loop adding stuff to an object.

IMHO `reduce` is unnecessarily complicated and doing `...acc` is not well optimized in JS; for (larger) namespaces of N translations, it's creating N objects, with 1-N property assignments. Plus, this code runs a lot.